// react-paginated/build/Paginate.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Context = _interopRequireDefault(require("./Context"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var Paginate = function Paginate(_ref) {
  var children = _ref.children,
      page = _ref.page,
      resultsPerPage = _ref.resultsPerPage,
      totalResults = _ref.totalResults,
      results = _ref.results,
      fragment = _ref.fragment,
      rest = _objectWithoutProperties(_ref, ["children", "page", "resultsPerPage", "totalResults", "results", "fragment"]);

  return _react.default.createElement(_Context.default.Provider, {
    value: {
      totalResults: totalResults,
      results: results,
      page: page,
      resultsPerPage: resultsPerPage,
      hasResults: Array.isArray(results) && results.length > 0
    }
  }, fragment ? children : _react.default.createElement("div", rest, children));
};

Paginate.defaultProps = {
  totalResults: 0
};
Paginate.propTypes = {
  children: _propTypes.default.node.isRequired,
  page: _propTypes.default.number.isRequired,
  resultsPerPage: _propTypes.default.number.isRequired,
  totalResults: _propTypes.default.number
};
var _default = Paginate;
exports.default = _default;
